specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json

```diff
@@ -0,0 +1,1275 @@
+      "additionalProperties": false,
+      "properties": {
+        "add": {
+          "type": "array",
+          "description": "List of questions to be added",
+          "maxLength": 100,
+          "items": {
+            "type": "string"
+          }
+        },
+        "delete": {
+          "type": "array",
+          "description": "List of questions to be deleted.",
+          "maxLength": 100,
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "UpdateMetadataDTO": {
+      "type": "object",
+      "description": "PATCH Body schema to represent list of Metadata to be updated",
+      "additionalProperties": false,
+      "properties": {
+        "delete": {
+          "type": "array",
+          "description": "List of Metadata associated with answer to be deleted",
+          "maxLength": 100,
+          "items": {
+            "$ref": "#/definitions/MetadataDTO"
+          }
+        },
+        "add": {
+          "type": "array",
+          "description": "List of Metadat associated with answer to be added",
+          "maxLength": 100,
+          "items": {
+            "$ref": "#/definitions/MetadataDTO"
+          }
+        }
+      }
+    },
+    "DeleteKbContentsDTO": {
+      "type": "object",
+      "description": "PATCH body schema of Delete Operation in UpdateKb",
+      "additionalProperties": false,
+      "properties": {
+        "ids": {
+          "type": "array",
+          "description": "List of Qna Ids to be deleted",
+          "items": {
+            "type": "integer",
+            "format": "int32"
+          }
+        },
+        "sources": {
+          "type": "array",
+          "description": "List of sources to be deleted from knowledgebase.",
+          "maxLength": 300,
+          "minLength": 1,
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "CreateKbInputDTO": {
+      "type": "object",
+      "description": "Input to create KB.",
+      "additionalProperties": false,
+      "properties": {
+        "qnaList": {
+          "type": "array",
+          "description": "list of QNA to be added to the index. Ids are generated by the service and should be omitted.",
+          "items": {
+            "$ref": "#/definitions/QnADTO"
+          }
+        },
+        "urls": {
+          "type": "array",
+          "description": "List of URLs to be added to knowledgebase.",
+          "maxLength": 10,
+          "items": {
+            "type": "string"
+          }
+        },
+        "files": {
+          "type": "array",
+          "description": "List of files to be added to knowledgebase.",
+          "maxLength": 10,
+          "items": {
+            "$ref": "#/definitions/FileDTO"
+          }
+        }
+      }
+    },
+    "KnowledgebaseEnvironment": {
+      "type": "string",
+      "description": "Enumeration of knowledgebase environments.",
+      "x-enumNames": [
```

Comment:
please take a look at https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum and use "modelAsString" : true if you think this list may expand later (to reduce breaking changes in generated SDKs)

Reply:
good suggestion